Source/Renderer/modernizeShader.js

```diff
@@ -99,21 +99,23 @@ function modernizeShader(source, isFragmentShader) {
     }
   }
 
+  var webgl2UniqueID = "WEBGL_2";
+  var webgl2DefineMacro = "#define " + webgl2UniqueID;
   var versionThree = "#version 300 es";
   var foundVersion = false;
   for (i = 0; i < splitSource.length; i++) {
     if (/#version/.test(splitSource[i])) {
-      splitSource[i] = versionThree;
+      splitSource[i] = versionThree + "\n" + webgl2DefineMacro;
       foundVersion = true;
     }
   }
 
   if (!foundVersion) {
-    splitSource.splice(0, 0, versionThree);
+    splitSource.splice(0, 0, versionThree, webgl2DefineMacro);
   }
```

Comment:
It might be simpler to just splice in the webgl2 macro after the version handling.


```js
  var webgl2UniqueID = "WEBGL_2";
  var webgl2DefineMacro = "#define " + webgl2UniqueID;
  var versionThree = "#version 300 es";
  var foundVersion = false;
  for (i = 0; i < splitSource.length; i++) {
    if (/#version/.test(splitSource[i])) {
      splitSource[i] = versionThree;
      foundVersion = true;
    }
  }

  if (!foundVersion) {
    splitSource.splice(0, 0, versionThree);
  }

  splitSource.splice(1, 0, webgl2DefineMacro);
```